### PR TITLE
Create User model

### PR DESCRIPTION
**Pull request by nomad-alt**
## PT link: https://www.pivotaltracker.com/story/show/151366010 ##
- Added devise gem
- Added User model

What we've learned: 
- How authentication mechanism works with Rails
- To create a working branch from the develop and not from the current working branch so that my last PR's commits are not displayed on my new PR.

### CODE REVIEW

**Comment by diraulo** [comment]
@nurlanemir we need to clean your commit history here... I suspect your `develop` branch is messed up.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (diraulo) [inline spec/models/user_spec.rb]: No tests for the user model? 🤔

**Review by diraulo — DISMISSED** [review]
Commit history looks good now but this PR has no tests at all? why?

File: features/step_definitions/user_registration_steps.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Given(/^the following users exists$/) do |table|
+  table.hashes.each do |hash|
+      User.create!(hash)
```
Comment (tochman) [inline features/step_definitions/user_registration_steps.rb]: 1. We don't use methods with a `!` unless there´s a really good reason. What would be the reason here? 
2. There's a gem called FactoryGirl installed. We used that gem in the Rails Messaging Legacy Code Challenge. Why not use it here?

File: features/user_registration.feature [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Feature: As a user in order to use the application
+    I should be able to sign up for account
```
Comment (tochman) [inline features/user_registration.feature]: The user story format has 3 components. This story only has 2. What is missing?

**Review by tochman — CHANGES_REQUESTED** [review]
You need to add more tests for the sad path (error handlers) and also unit specs for the user.

File: app/controllers/user_controller.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class UserController < ApplicationController
+  def new
+  end
+end
```
Comment (tochman) [inline app/controllers/user_controller.rb]: Why is this controller created and why do you have a `new`method? Try running your tests without it.

**Review by tochman — CHANGES_REQUESTED** [review]
I don't see a reason why you have generated all the views for devise. I will let it pass but there is no need to do that at this stage. The authentication would have worked anyway. (See my other comments)

**Review by tochman — APPROVED** [review]

**Pull request merged**